### PR TITLE
feat: add slider limits to web

### PR DESCRIPTION
**Pull request by dcangulo**
Summary:
---------

Add `upperLimit` support and **partial** `lowerLimit` support to web.

While `upperLimit` works perfectly, I can't seem to figure out how to calculate the `lowerLimit` offset but I included it anyway even though I only need `upperLimit` support, maybe someone else can improve the partial support, any guidance is appreciated.

Test Plan:
----------
## No `upperLimit` and `lowerLimit`
```js
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import Slider from '@react-native-community/slider';

export default function App() {
  const [value, setValue] = React.useState(5);

  return (
    <View style={styles.container}>
      <View>
        <Text>Current Value: {value}</Text>
        <Slider
          style={{ width: 400, height: 40 }}
          value={value}
          minimumValue={1}
          maximumValue={10}
          step={1}
          onValueChange={setValue}
        />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
```
https://github.com/callstack/react-native-slider/assets/36528176/b5453064-4bf6-4f13-a6cf-92a1ffd1522e

-----
## `upperLimit` = 8, no `lowerLimit`
```js
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import Slider from '@react-native-community/slider';

export default function App() {
  const [value, setValue] = React.useState(5);

  return (
    <View style={styles.container}>
      <View>
        <Text>Current Value: {value}</Text>
        <Slider
          style={{ width: 400, height: 40 }}
          value={value}
          minimumValue={1}
          maximumValue={10}
          step={1}
          upperLimit={8}
          onValueChange={setValue}
        />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
```
https://github.com/callstack/react-native-slider/assets/36528176/8525c645-1b7d-4e83-a5b8-b09bb2131ecd

----
## `upperLimit` = 8, `lowerLimit` = 2
```js
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import Slider from '@react-native-community/slider';

export default function App() {
  const [value, setValue] = React.useState(5);

  return (
    <View style={styles.container}>
      <View>
        <Text>Current Value: {value}</Text>
        <Slider
          style={{ width: 400, height: 40 }}
          value={value}
          minimumValue={1}
          maximumValue={10}
          step={1}
          lowerLimit={2}
          upperLimit={8}
          onValueChange={setValue}
        />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
```
https://github.com/callstack/react-native-slider/assets/36528176/5b6df434-09bf-4eda-988c-6c6e3e612c44

----
## no `upperLimit`, `lowerLimit` = 2
```js
import React from 'react';
import { StyleSheet, Text, View } from 'react-native';
import Slider from '@react-native-community/slider';

export default function App() {
  const [value, setValue] = React.useState(5);

  return (
    <View style={styles.container}>
      <View>
        <Text>Current Value: {value}</Text>
        <Slider
          style={{ width: 400, height: 40 }}
          value={value}
          minimumValue={1}
          maximumValue={10}
          step={1}
          lowerLimit={2}
          onValueChange={setValue}
        />
      </View>
    </View>
  );
}

const styles = StyleSheet.create({
  container: {
    flex: 1,
    backgroundColor: '#fff',
    alignItems: 'center',
    justifyContent: 'center',
  },
});
```
https://github.com/callstack/react-native-slider/assets/36528176/7989e4a4-f783-48a7-88f0-f27274b7dc18

---

As you can see in the demo, the larger the `lowerLimit`, the larger the gap between the `thumb` and the `cursor`.

### CODE REVIEW

File: package/src/RNCSliderNativeComponent.web.tsx [hunk 1]
```diff
@@ -31,6 +31,8 @@ export interface Props {
   value: number;
   minimumValue: number;
   maximumValue: number;
+  lowerLimit: number;
+  upperLimit: number;
```
Comment (BartoszKlonowski) [inline package/src/RNCSliderNativeComponent.web.tsx:35]: These definitely should be optional.

File: package/src/RNCSliderNativeComponent.web.tsx [hunk 2]
```diff
@@ -318,8 +324,10 @@ const RCTSliderWebComponent = React.forwardRef(
       <View
         ref={containerRef}
         onLayout={({nativeEvent: {layout}}: LayoutChangeEvent) => {
+          const lowerValue = lowerLimit === minimumValue ? 0 : lowerLimit;
```
Comment (BartoszKlonowski) [inline package/src/RNCSliderNativeComponent.web.tsx:327]: I'm not sure if I understand this implementation - we put `0` if `lowerLimit` is equal to `minimumValue`?
Why? What if user set's `lowerLimit={10}` and `minimumValue={10}`? We would still be putting 0 here?

File: package/src/RNCSliderNativeComponent.web.tsx [hunk 3]
```diff
@@ -318,8 +324,10 @@ const RCTSliderWebComponent = React.forwardRef(
...
+          const multiplier = (upperLimit - lowerValue) / maximumValue;
           containerSize.current.height = layout.height;
-          containerSize.current.width = layout.width;
+          containerSize.current.width = layout.width * multiplier;
```
Comment (BartoszKlonowski) [inline package/src/RNCSliderNativeComponent.web.tsx:330]: Sliding limits definitely shouldn't affect the overall layout and/or look of the Slider. They should only prevent user from sliding above the given limits, indicating there's still some value unreachable to user.
But with this implementation we manipulate with the length of the sliding track, why?

**Review by BartoszKlonowski — COMMENTED** [review]
Thanks for taking the effort to implement that!
I very briefly went through the implementation and have some questions.

I will (on Friday, 14.07) go through the implementation deeper also trying to fix the missing functionality you mentioned.

**Comment by JacobDel** [comment]
Please close this old PR.
New PR request: https://github.com/callstack/react-native-slider/pull/565

**Pull request closed without merging**